src/magic_uv/op/texture_wrap.py

```diff
@@ -238,26 +227,65 @@ def execute(self, context):
+
+                # AB = shared edge, C = third vert of ref face
+                # set up a 2D coordinate system with coordinates relative to AB
+                # X = C projected onto AB, XC/AX = perpendicular/parallel to AB
+                xc_3d, x_3d = common.diff_point_to_segment(a_3d, b_3d, c_3d)
+                ax_3d = x_3d - a_3d
+                ab_3d = b_3d - a_3d
+                ab_2d = Vector((0.0, ab_3d.length))
+                ac_2d = Vector((xc_3d.length,
+                                math.copysign(ax_3d.length, ax_3d.dot(ab_3d))))
+                ab_uv = b_uv - a_uv
+                ac_uv = c_uv - a_uv
+
+                # extra check for collinear verts
+                if xc_3d.length < 1e-5:
+                    continue
+
+                # find affine transformation from this 2D system to UV
+                #  [u] = [m11 m12] @ [x]
+                #  [v]   [m21 m22]   [y]       [u1]   [x1 y1 0  0 ]   [m11]
+                #                              [v1] = [0  0  x1 y1] @ [m12]
+                #  u = m11*x + m12*y           [u2]   [x1 y1 0  0 ]   [m21]
+                #  v = m21*x + m22*y           [v2]   [0  0  x1 y1]   [m22]
+                vector_uv = Vector((ab_uv.x, ab_uv.y, ac_uv.x, ac_uv.y))
+                matrix_2d = Matrix(((ab_2d.x, ab_2d.y, 0, 0),
+                                    (0, 0, ab_2d.x, ab_2d.y),
+                                    (ac_2d.x, ac_2d.y, 0, 0),
+                                    (0, 0, ac_2d.x, ac_2d.y)))
+                try:
+                    m_coeffs = linalg.solve(matrix_2d, vector_uv)
+                    tform_mtx = Matrix(((m_coeffs[0], m_coeffs[1]),
+                                        (m_coeffs[2], m_coeffs[3])))
+                    break
+                except linalg.LinAlgError:
+                    pass    # loop and try a different third vert
+
+            if tform_mtx is None:
+                self.report({'WARNING'}, "Invalid reference face")
+                return {'CANCELLED'}
+
+            # find UVs for target vertices
             for info in tgt_other_verts:
-                cv0 = common_verts[0]["vert"].co
-                cv1 = common_verts[1]["vert"].co
-                cuv0 = common_verts[0]["ref_loop"][uv_layer].uv
-                ov = info["vert"].co
-                info["vert_hdiff"], x = common.diff_point_to_segment(
-                    cv0, cv1, ov)
-                info["vert_vdiff"] = x - common_verts[0]["vert"].co
-
-                # calclulate factor
-                fact_h = -info["vert_hdiff"].length / \
-                    ref_info["vert_hdiff"].length
-                fact_v = info["vert_vdiff"].length / \
-                    ref_info["vert_vdiff"].length
-                duv_h = ref_info["uv_hdiff"] * fact_h
-                duv_v = ref_info["uv_vdiff"] * fact_v
-
-                # get target UV
-                info["target_uv"] = cuv0 + duv_h + duv_v
+                d_3d = info["vert"].co
```

Comment:
Could you also add the comment here like L240-L242 for the explanation of the algorithm?